lib/internal/process/warning.js

```diff
@@ -116,24 +116,23 @@ function setupProcessWarnings() {
       ctor = type;
       code = undefined;
       type = 'Warning';
+    } else if (type !== undefined && typeof type !== 'string') {
```

Comment:
I don't think this is the same. If `type` is an object but `type.type` is truthy and not a `string` then this won't throw anymore.

Reply:
Addressed.